### PR TITLE
Configure names for private listeners

### PR DESCRIPTION
**Pull request by rgrandl**
Before this PR, the user is able to specify a hostname for the public listeners. However, for the private listeners, we compute an internal name by default. This is not great, especially if you want to change the names of the internal listeners (e.g., deploy in different environments).

This PR changes the way the users can configure the listeners: 1) If the user specify a listener config, it has to specify whether the
   listener is public by setting `is_public` knob explicitly.
2) If `is_public` knob is not set, the listener will be private by
   default.
3) For public listeners, the user is required to specify the `hostname`. 4) For private listeners, `hostname` is optional; if not set, we'll
   create a default name:
   <listener_name>.<region>.serviceweaver.internal; if set, the name
   will be:
   <hostname>.<listener_name>.<region>.serviceweaver.internal

### CODE REVIEW

File: internal/config/config.proto [hunk 1]
```diff
@@ -43,12 +43,14 @@ message GKEConfig {
   // Options for the application listeners, keyed by listener name.
   // If a listener isn't specified in the map, default options will be used.
   message ListenerOptions {
-    // Public hostname for the listener. If empty, the listener is assumed
-    // to be private.
+    // If true, the listener is assumed to be public.
```
Comment (spetrovic77) [inline internal/config/config.proto:46]: Rewrite to:

```
// Is the listener public, i.e., should it be accessible from outside of the cloud project?
```

**Review by spetrovic77 — APPROVED** [review]
Nice, very clean!

File: internal/config/config.proto (hunk 1 above)
Comment (rgrandl) [reply internal/config/config.proto]: Done

**Review by rgrandl — COMMENTED** [review]
Thanks, Srdjan.

**Pull request merged**